ref.yml

```diff
@@ -392,29 +399,196 @@ fields:
       used:
         - coq.opam
         - .travis.yml
-  - tested_coq_nix_versions:
+  - cachix:
+      used:
+        - nix-action.yml.nix
       type: list
-      description: If empty, the master branch of Coq will be tested.
+      description: Cachix repositories to use
       item_fields:
-        - version_or_url:
+        - name:
             required: true
-            description: >
-              This can be a version number of a version packaged in
-              nixpkgs (example: 8.5, 8.11).  Cf. the coqPackages_8_XX
-              attributes defined at the end of this file:
-              https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/top-level/coq-packages.nix
-              (Quote the version number, otherwise '8.10' becomes '8.1'.)
-
-              Or this can be a URL, to test your project with a
-              version of Coq under development.  Typically this is
-              https://github.com/coq/coq-on-cachix/tarball/master but
-              'master' may be replaced by any valid reference like one
-              of the branches available at
-              https://github.com/coq/coq-on-cachix/branches or a valid
-              commit hash.
+            description: Name of the cachix repository
+            examples:
+              - "coq"
+              - "math-comp"
+        - key:
+            required: false
+            description: |
+              Secret signingKey for the repository.
+              The secret should be generated from https://cachix.org
+              and added to Github, either:
+              - at the repository level
+                https://github.com/OWNER/REPO/settings/secrets/actions
+              - or at the organization level:
+                https://github.com/organizations/ORG/settings/secrets/actions
+              /!\ This is the legacy authentification method and
+              /!\ is discouraged in favor of `token` below.
+            example:
+              - "CACHIX_SIGNING_KEY"
+        - token:
```

Comment:
Note that we could set a default value for this `token` if `community` is `true`.

Reply:
FWIW, I tried with this patch but it didn't work:

```diff
diff --git a/nix-action.yml.mustache b/nix-action.yml.mustache
index 3f67a5c..60a4e9f 100644
--- a/nix-action.yml.mustache
+++ b/nix-action.yml.mustache
@@ -39,6 +39,21 @@ jobs:
 <%# token%>
         authToken: '${{ secrets.<%token%> }}'
 <%/ token%>
+<%/ cachix%>
+<%^ cachix%>
+<%# community%>
+    - name: Cachix setup coq
+      uses: cachix/cachix-action@v8
+      with:
+        # Name of a cachix cache to pull/substitute
+        name: coq
+    - name: Cachix setup coq-community
+      uses: cachix/cachix-action@v8
+      with:
+        # Name of a cachix cache to pull/substitute
+        name: coq-community
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+<%/ community%>
 <%/ cachix%>
     - name: Git checkout
       uses: actions/checkout@v2
@@ -87,6 +102,21 @@ jobs:
 <%# token%>
         authToken: '${{ secrets.<%token%> }}'
 <%/ token%>
+<%/ cachix%>
+<%^ cachix%>
+<%# community%>
+    - name: Cachix setup coq
+      uses: cachix/cachix-action@v8
+      with:
+        # Name of a cachix cache to pull/substitute
+        name: coq
+    - name: Cachix setup coq-community
+      uses: cachix/cachix-action@v8
+      with:
+        # Name of a cachix cache to pull/substitute
+        name: coq-community
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+<%/ community%>
 <%/ cachix%>
     - name: Git checkout
       uses: actions/checkout@v2
```

My Mustache is too rusty for me to understand what the problem is.

Reply:
I did manage to do something similar with default Cachix instances and default Cachix pushing for coq-community projects in #99.